compose/core-compose/src/main/java/com/squareup/workflow/ui/compose/RenderAsState.kt

```diff
@@ -220,11 +220,13 @@ private class WorkflowState<PropsT, OutputT : Any, RenderingT>(
     propsChannel.offer(props)
   }
 
-  override fun onEnter() {}
+  override fun onRemembered() {}
 
-  override fun onLeave() {
+  override fun onForgotten() {
     workflowScope.cancel()
   }
+
+  override fun onAbandoned() {}
```

Comment:
Should also cancel the scope here, since this means the composition was unsuccessful. This was actually a latent bug (#297), so I'm really excited that this new API allows us to fix it. 
```suggestion
  override fun onAbandoned() {
    workflowScope.cancel()
  }
```

Then add "Fixes #297" to the commit message please.